README.md

```diff
@@ -0,0 +1,142 @@
+
+## [Pipeline](webidls/Pipeline.webidl)
+
+A page.
+
+A reference to a servo pipeline. Offers access to many internal information
+about the document. Fires many events to keep track of the document status
+(loading state, security state, etc) and request user actions (prompts,
+security questions, …).
+
+Can be pending, loading, interactive, complete (loaded).
+
+A pipeline implements some extra interfaces: [Editable](webidls/Editable.webidl)
+(to build the app "edit" menu), [FindInPage](webidls/FindInPage.webidl) (to build
+a in page text search), [HTTPObserverManager](webidls/HTTPObserverManager.webidl)
+(to track and overwrite HTTP connections), [MultimediaManager](webidls/MultimediaManager.webidl)
+(to track multimedia content, to silence a tab for example),
+[Printable](webidls/Printable.webidl) (to implement Print to printer or Print to PDF).
+
+To preloada page (first result in the urlbar for example),
+it's possible to create a new "orphan" pipeline and later attach it to a
+Browser, as long as they share the same session.
+
+## [LoadData](webidls/LoadData.webidl)
+
+Serializable. Minimal set of information to create or restore a pipeline. Is
+used to save a history entry if pipeline is being purged (#11893). Is used to
+restore session (list of LoadData can be store on disk). Is used to transmit
+request, to the embedder, to open a new window or a new tab.
+
+## [StorageSession](webidls/Session.webidl) 
+
+The StorageSession object, and restore session.
+
+Holds offline data: appcache, cookies, fileSystem, indexdb, localStorage,
+serviceworkers. Has methods to clear data. Is serializable to write to disk.
+
+No disk IO is done in Servo.
+
+Up to the embedder to save and restore data from disk.
+
+A storage session stores offline data (cookies, localStorage, …). To build a
+session recovery, storage session needs to be serialized and written to disk,
+along with a list of list of list of LoadData (for tab restore). List of list
+of list because a browser is usually made of a list of windows made of a list
+of tabs made of a list of history entries. It's up to the embedder to
+regularly saved the session. The process of writing the storage session and
+the loaddata will require disk access, which is supposed to be handled by the
+embedder.
+
+## [ContentBlockers](webidls/ContentBlockers.webidl)
+
+A Browser has access to multiple content blockers: Popup blocker, tracking
+content blocker, mixed content blocker, custom blocker (à la Safari). A
+pipeline can temporarly enable/disable a content blocker.
+
+## [Viewport](webidls/Viewport.webidl)
+
+Where a pipeline is rendered.
+
+We want to dissociate clipping area and content boundaries. A viewport object
+defines the boundaries geometry.
+
+A viewport is responsible to pass keyboard/mouse/touch events to the
+Browser/pipeline.
+
+3 types of viewport:
+- interactive: its layout define the size of the rendered pipeline. Events
+  are forwarded to the pipeline.
```

Comment:
It would be helpful to know why forwarding the size info to a pipeline make more sense than just manually reflecting size of the pipeline.